PHPCompatibility/Tests/Constants/TemporarilyRemovedConstantsUnitTest.inc

```diff
@@ -0,0 +1,4 @@
+<?php
+
+// Constants which were present, then removed, then present again.
+echo T_BAD_CHARACTER;
```

Comment:
Just to be on the safe site (and to actually safeguard the logic in the sniff), what about adding some extra tests like these ?
```php
echo t_bad_character; // Not the uppercase constant.
echo BAD_CHARACTER; // Not the right constant.
echo My\NS\T_BAD_CHARACTER; // Not the global constant.
echo self::T_BAD_CHARACTER; // Not the global constant.
echo $obj->T_BAD_CHARACTER; // Not a constant.
function T_BAD_CHARACTER() {} // Not a constant.
```

Those would need to be tested with a `testNoFalsePositives($line)` method against a `testVersion` on which the sniff would normally throw an error.

Reply:
This diverges from the other two related sniffs. Would you like the same to be applied there too?

Reply:
Aren't those type of tests in those sniff test case files at all ? Hmm.. there should be some.

Also just thought of another test case:
```php
echo \T_BAD_CHARACTER; // This is the global constant and should be flagged.
```